File: student.js
var ename=["sno","firstname","lastname","courses","status","semester",]
function insertData(){

var jsondata = {
	"sno":document.getElementById("sno").value,
	"firstname":document.getElementById("firstname").value,
	"lastname":document.getElementById("lastname").value,
	"status":document.getElementById("status").value,
}
var radioelement = document.querySelector('input[name="semester"]:checked')
if(radioelement!=null){
	jsondata['semester']=radioelement.value;
}else{
	jsondata['semester']=''
}

var checkbox=[];
var checked1=document.querySelectorAll('input[type="checkbox"]:checked');
for(var e of checked1){
	checkbox.push(e.value);
}
jsondata["courses"]=checkbox;
var url='http://localhost:5000/webforms/insert/'
    $.ajax({
        url:url,
        type:"POST",
        data: JSON.stringify(jsondata),
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        beforeSend: function(request) {
            request.setRequestHeader("Access-Control-Allow-Origin","*")
            request.setRequestHeader("Access-Control-Allow-Methods","POST");
        },
        success: function(response){
            if(response.ok){
                document.getElementById("result").innerHTML="<p>Data:"+JSON.stringify(response.data)+"  is inserted in database</p>";
                alert("Data:"+JSON.stringify(response.data)+"  is inserted in database");
                console.log(response);
            }else{
                alert(response.error);
            }
        },
        error: function(error){
                alert("ERROR "+JSON.stringify(error))
            },
    }).fail(function (jqXHR, textStatus, errorThrown) {
        alert("Connection Refused, Please try again!")
    });
    
}

function displayContent(){
    var dtable = document.createElement('table');
    dtable.setAttribute('id','dtable');
    dtable.setAttribute('class','table');
    dtable.setAttribute('border',2);
    var tr = dtable.insertRow(0);
    ename.forEach(function(e){
        var th = document.createElement('th');
        th.innerHTML = e;
        tr.appendChild(th);
    });
    
var url =  'http://localhost:5000/webforms/display/'
        $.ajax({
        url: url,
        type: 'GET',
        beforeSend: function(request) {
            request.setRequestHeader("Access-Control-Allow-Origin","*")
            request.setRequestHeader("Access-Control-Allow-Methods","GET");
        },
        success: function(response){
            if(response.ok){
                Object.keys(response.data).forEach(function(key){
                var value = response.data[key];
                console.log(key)
                document.getElementById("display_table").innerHTML+="<p><b>"+key+"</b></p>";
                var dtable = document.createElement('table');
                    dtable.setAttribute('id',key);
                    dtable.setAttribute('class','table');
                    dtable.setAttribute('border',2);
                    dtable.setAttribute('style','width:auto');
                    var tr = dtable.insertRow(0);
                Object.keys(value[0]).forEach(function(head){
                    var th = document.createElement('th');
                    th.innerHTML = head;
                    tr.appendChild(th);
                
                })
                i=1
                for(var v of value){
                    var tr = dtable.insertRow(i);
                    i+=1;
                    Object.keys(v).forEach(function(innerKey){
                        var td = document.createElement('td');
                        td.innerHTML = v[innerKey];
                        tr.appendChild(td); 
                    })
                }
                    var div = document.getElementById('display_table');
                    div.appendChild(dtable);
                
                }) 
            }else{
                alert(response.error);
            }
        },
        error: function(error){
                alert("ERROR "+JSON.stringify(error))
            },
    }).fail(function (jqXHR, textStatus, errorThrown) {
        alert("Connection Refused, Please try again!")
    });
    document.getElementById("display").disabled = 'true';
}
    

File: student.py
from flask import Flask, jsonify
from flask import abort
from flask import make_response
from flask import request
import mysql.connector as mysql
from flask_cors import CORS

app = Flask(__name__)
CORS(app)


@app.route ('/webforms/insert/', methods=['POST'])
def insert():
	sql1 = "INSERT INTO student values(" + request.json['sno'] + ",'" + request.json['firstname'] + "','" + request.json['lastname'] + "','" + request.json['status'] + "','" + request.json['semester'] + "')"

	db = mysql.connect(
	 host="localhost",
	 database="project3",
	 user="root",
	 passwd="Adventure@25",
	 auth_plugin='mysql_native_password'
	)
	cursor = db.cursor(dictionary=True)
	try:
		cursor.execute(sql1)
		for l in request.json['courses']:
			sql2 = "INSERT INTO courses values(" + request.json['sno'] + ",'" + l + "')"
			cursor.execute(sql2)

		db.commit()
		cursor.close()
		db.close()
		result = {"ok": True,"data":request.json}
		return jsonify(result)
	except Exception as e:
		db.rollback()
		cursor.close()
		db.close()
		result = {"ok": False,"error":e.__str__()}
		return jsonify(result)


@app.route('/webforms/display/',methods=['GET'])
def get_student():
	db = mysql.connect(
	 host="localhost",
	 database="project3",
	 user="root",
	 passwd="Adventure@25",
	 auth_plugin='mysql_native_password'
	)
	query1 = "select sno,firstname,lastname,status,semester from student"
	cursor = db.cursor(dictionary=True)
	try:
		cursor.execute(query1)
		records = cursor.fetchall()
		student=[]
		for record in records:
			student.append({'sno':record['sno'],'firstname':record['firstname'],'lastname':record['lastname'],'status':record['status'],'semester':record['semester']})
		courses=[]

		query2 = "select * from courses"
		cursor.execute(query2)
		record2 = cursor.fetchall()

		for record in record2:
			courses.append({'sno':record['sno'],'courses_value':record['courses_value']})
		data = {'student':student,'courses':courses}
		result = {"ok": True,"data":data}
		return jsonify(result)
	except Exception as e:
		cursor.close()
		db.close()
		result = {"ok": False,"error":e.__str__()}
		return jsonify(result)

if __name__== '__main__':
	app.run(host='localhost',debug=True)